packages/core/src/config/parse.ts

```diff
@@ -1058,11 +1136,18 @@ const parseContractVariables = (
  * @returns complete set of variables parsed into the format expected by the parsed chugsplash config.
  */
 export const parseContractConstructorArgs = (
-  userConstructorArgs: UserConfigVariables,
+  userContractConfig: UserContractConfig,
   referenceName: string,
   abi: Array<Fragment>,
   cre: ChugSplashRuntimeEnvironment
 ): ParsedConfigVariables => {
+  let userConstructorArgs: UserConfigVariables =
+    userContractConfig.constructorArgs ?? {}
+
+  userConstructorArgs = JSON.parse(
+    Handlebars.compile(JSON.stringify(userConstructorArgs))({})
+  )
```

Comment:
I've intentionally included the handlebars compilation here even though this PR does not include contract reference resolution. This is because performing the compilation meaningfully affects the output type metadata and the variable validation logic I am reusing assumes that the compilation has already been run. So it just simplifies things to include it here even though it's not strictly required yet.